x-pack/platform/plugins/private/reporting/public/management/components/reporting_tabs.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiLoadingSpinner, EuiPageTemplate } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { Route, Routes } from '@kbn/shared-ux-router';
+import { RouteComponentProps } from 'react-router-dom';
+import { CoreStart, ScopedHistory } from '@kbn/core/public';
+import { ILicense, LicensingPluginStart } from '@kbn/licensing-plugin/public';
+import { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import {
+  ClientConfigType,
+  ReportingAPIClient,
+  useInternalApiClient,
+  useKibana,
+} from '@kbn/reporting-public';
+import { SharePluginStart } from '@kbn/share-plugin/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import useObservable from 'react-use/lib/useObservable';
+import { Observable } from 'rxjs';
+import { SCHEDULED_REPORT_VALID_LICENSES } from '@kbn/reporting-common';
+import { suspendedComponentWithProps } from './suspended_component_with_props';
+import { REPORTING_EXPORTS_PATH, REPORTING_SCHEDULES_PATH, Section } from '../../constants';
+import ReportExportsTable from './report_exports_table';
+import { IlmPolicyLink } from './ilm_policy_link';
+import { ReportDiagnostic } from './report_diagnostic';
+import { useIlmPolicyStatus } from '../../lib/ilm_policy_status_context';
+import { MigrateIlmPolicyCallOut } from './migrate_ilm_policy_callout';
+import ReportSchedulesTable from './report_schedules_table';
+import { LicensePrompt } from './license_prompt';
+
+export interface MatchParams {
+  section: Section;
+}
+
+export interface ReportingTabsProps {
+  coreStart: CoreStart;
+  license$: LicensingPluginStart['license$'];
+  dataService: DataPublicPluginStart;
+  shareService: SharePluginStart;
+  config: ClientConfigType;
+  apiClient: ReportingAPIClient;
+}
+
+export const ReportingTabs: React.FunctionComponent<
+  Partial<RouteComponentProps> & ReportingTabsProps
+> = (props) => {
+  const { coreStart, license$, shareService, config, ...rest } = props;
```

Comment:
Can we get some of these services from `useKibana`? If so, they don't need to be passed as props.

Reply:
We usually do this with data-fetching hooks because if they need to be reused and thus are moved to packages, then we cannot rely on the Kibana context anymore. It's unlikely that these will be ever reused, but we tend to do this anyway just in case